source/extensions/filters/http/cors/cors_filter.cc

```diff
@@ -200,11 +205,37 @@ bool CorsFilter::allowCredentials() {
   return false;
 }
 
+std::string CorsFilter::runtimeKey(const std::string& key) {
+  for (const auto policy : policies_) {
+    if (!policy) {
+      continue;
+    }
+    if (!policy->runtimeKey().empty()) {
+      return fmt::format("cors.{}.{}", policy->runtimeKey(), key);
+    }
+  }
+  return EMPTY_STRING;
+}
+
+bool CorsFilter::shadowEnabled() {
+  const auto shadow_mode = runtimeKey("shadow_enabled");
+  if (shadow_mode == EMPTY_STRING) {
+    return false;
+  }
+  return config_->runtime().snapshot().featureEnabled(shadow_mode, 0);
+}
+
 bool CorsFilter::enabled() {
+  const auto filter_enabled = runtimeKey("filter_enabled");
   for (const auto policy : policies_) {
-    if (policy) {
-      return policy->enabled();
+    if (!policy) {
+      continue;
+    }
+    bool policy_enabled = policy->enabled();
+    if (filter_enabled == EMPTY_STRING) {
+      return policy_enabled;
     }
+    return config_->runtime().snapshot().featureEnabled(filter_enabled, int(policy_enabled) * 100);
```

Comment:
`policy->enabled()` can be inlined and the first `if` can be dropped as discussed earlier.

Reply:
I need the first conditional otherwise the return of runtime would happen even if a policy isn't defined. e.g.
```cc
  for (const auto policy : policies_) {
    if (policy && filter_enabled == EMPTY_STRING) {
      return policy->enabled();
    }
    // The next line will execute even if there is no policy
    return config_->runtime().snapshot().featureEnabled(filter_enabled, int(policy->enabled()) * 100);
  }
```

Reply:
Oh, yea, I meant to say that we can combine both.

Reply:
something like this?
```cc
  for (const auto policy : policies_) {
    if (policy) {
      if (filter_enabled == EMPTY_STRING) {
        return policy->enabled();
      }
      return config_->runtime().snapshot().featureEnabled(filter_enabled, int(policy->enabled()) * 100);
    }
  }
```

Reply:
Never mind, you can leave this as-is.